pkg/apis/resource/validation/validation_resourceclaimtemplate_test.go

```diff
@@ -307,7 +308,10 @@ func TestValidateClaimTemplateUpdate(t *testing.T) {
 		t.Run(name, func(t *testing.T) {
 			scenario.oldClaimTemplate.ResourceVersion = "1"
 			errs := ValidateClaimTemplateUpdate(scenario.update(scenario.oldClaimTemplate.DeepCopy()), scenario.oldClaimTemplate)
-			assert.Equal(t, scenario.wantFailures, errs)
+			assert.Equal(t, len(scenario.wantFailures), len(errs))
+			for i := 0; i < len(errs); i++ {
+				assert.True(t, strings.Contains(errs[i].Detail, scenario.wantFailures[i].Detail))
```

Comment:
`assert.True` alone won't tell us which index failed unless we include more details. Something like this will add details if we fail

```suggestion
				assert.True(t, strings.Contains(errs[i].Detail, scenario.wantFailures[i].Detail), "expected error containing '%s', got '%s'", scenario.wantFailures[i].Detail, errs[i].Detail)
```

same comment other places this is just calling `assert.True(t, <bool>)`

Reply:
Or use https://pkg.go.dev/github.com/stretchr/testify/assert#Contains, then assert has access to the actual string and the expected sub-string and can produce a proper error message when the assertion fails.

What's then still unclear in case of a failure is the value of "i". This can be added with the additional message:

```suggestion
				assert.Contains(t, errs[i].Detail, scenario.wantFailures[i].Detail, "Error #%d", i)
```

Reply:
Thanks a lot! changes have been made accordingly